File: fixtures/typescript/src/dog.ts
/**
 * 狗类 - 继承自动物
 */
import { Animal } from './animal';

export class Dog extends Animal {
    private breed: string;
    
    constructor(name: string, age: number, breed: string) {
        super(name, age);
        this.breed = breed;
    }
    
    makeSound(): void {
        console.log(`${this.name} says: Woof! Woof!`);
    }
    
    fetch(): void {
        console.log(`${this.name} is fetching the ball`);
    }
    
    getBreed(): string {
        return this.breed;
    }
    
    toString(): string {
        return `Dog(name=${this.name}, age=${this.age}, breed=${this.breed})`;
    }
}

File: fixtures/python/dog.py
"""
狗类 - 继承自动物
"""
from animal import Animal


class Dog(Animal):
    """狗类"""
    
    def __init__(self, name: str, age: int, breed: str):
        super().__init__(name, age)
        self.breed = breed
    
    def make_sound(self) -> None:
        """狗叫"""
        print(f"{self.name} says: Woof! Woof!")
    
    def fetch(self) -> None:
        """捡球"""
        print(f"{self.name} is fetching the ball")
    
    def get_breed(self) -> str:
        return self.breed
    
    def __str__(self) -> str:
        return f"Dog(name={self.name}, age={self.age}, breed={self.breed})"
